code/game/say.dm

```diff
@@ -111,24 +118,38 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	if(copytext_char(input, -2) == "!!")
 		spans |= SPAN_YELL
 
+	var/say_mod = message_mods[MODE_CUSTOM_SAY_EMOTE]
+	if(!say_mod)
+		say_mod = say_mod(input, message_mods)
+
 	var/spanned = attach_spans(input, spans)
-	return "[say_mod(input, message_mods)], \"[spanned]\""
+	return "[say_mod], \"[spanned]\""
+
+/// Transforms the speech emphasis mods from [/atom/movable/proc/say_emphasis] into the appropriate HTML tags. Includes escaping.
+#define ENCODE_HTML_EMPHASIS(input, char, html, varname) \
+	var/static/regex/##varname = regex("(?<!\\\\)[char](.+?)(?<!\\\\)[char]", "g");\
+	input = varname.Replace_char(input, "<[html]>$1</[html]>")
+
+/// Scans the input sentence for speech emphasis modifiers, notably |italics|, +bold+, and _underline_ -mothblocks
+/atom/proc/say_emphasis(input)
+	ENCODE_HTML_EMPHASIS(input, "\\|", "i", italics)
+	ENCODE_HTML_EMPHASIS(input, "\\+", "b", bold)
+	ENCODE_HTML_EMPHASIS(input, "_", "u", underline)
```

Comment:
Also I don't think we should allow people to talk in underlines, since bold provides the emphasis that underlined would need and it kind of ends up looking more like a hyperlink than some important text.
![image](https://user-images.githubusercontent.com/26465327/204843996-b8cf0400-31d0-44cb-8938-8006a2d4a91e.png)
If other maintainers disagree with me (I've asked in dev channel), then I'll resolve this.

Reply:
Underline has been removed